examples/sprite_batch/Batching.cpp

```diff
@@ -0,0 +1,265 @@
+
+    int currentTex = 0;
+    while (tempTexture.loadFromFile(resourcesDir() / ("Tex" + std::to_string(currentTex) + ".png")))
+    {
+        textures.push_back(std::move(tempTexture));
+        tempTexture = sf::Texture();
+        currentTex += 1;
+    }
+
+    std::vector<sf::Sprite> sprites;
+
+    std::size_t spriteCount = 100;
+
+    sf::Font font;
+    if (!font.loadFromFile(resourcesDir() / "tuffy.ttf"))
+        return EXIT_FAILURE;
+
+    sf::Text text;
+    text.setOutlineColor(sf::Color::Blue);
+    text.setOutlineThickness(4);
+    text.setFillColor(sf::Color::White);
+    text.setFont(font);
+
+    sf::SpriteBatch            batch;
+    sf::SpriteBatch::BatchMode batchMode = sf::SpriteBatch::BatchMode::Immediate;
+
+    std::vector<float> timeHistory = {};
+
+    sf::Time    timeReference;
+    std::size_t framesTillRefUpdate = 0;
+    sf::Clock   clock;
+    while (window.isOpen())
+    {
+        // Handle events
+        for (sf::Event event; window.pollEvent(event);)
+        {
+            const int spriteIncrease = 100;
+
+            // Window closed or escape key pressed: exit
+            if ((event.type == sf::Event::Closed) ||
+                ((event.type == sf::Event::KeyPressed) && (event.key.code == sf::Keyboard::Escape)))
+            {
+                window.close();
+                break;
+            }
+
+            // Space key pressed: play
+            if (((event.type == sf::Event::KeyPressed) && (event.key.code == sf::Keyboard::W)))
+            {
+                spriteCount += spriteIncrease;
+            }
+
+            if (((event.type == sf::Event::KeyPressed) && (event.key.code == sf::Keyboard::S)))
+            {
+                spriteCount = std::max(spriteCount - spriteIncrease, minSprites);
+            }
+
+            if (((event.type == sf::Event::KeyPressed) &&
+                 (event.key.code == sf::Keyboard::F || event.key.code == sf::Keyboard::G)))
+            {
+                static sf::SpriteBatch::BatchMode modes[] = {
+                    sf::SpriteBatch::BatchMode::Immediate,
+                    sf::SpriteBatch::BatchMode::Deferred,
+                    sf::SpriteBatch::BatchMode::TextureSort,
+                    sf::SpriteBatch::BatchMode::DepthSort,
+                };
+
+                static std::size_t modeCount = sizeof(modes) / sizeof(sf::SpriteBatch::BatchMode);
+
+                for (std::size_t i = 0; i < modeCount; i++)
+                {
+                    if (batchMode == modes[i])
+                    {
+                        if (event.key.code == sf::Keyboard::F)
+                        {
+                            batchMode = modes[i == 0 ? modeCount - 1 : i - 1];
+                        }
+                        else
+                        {
+                            batchMode = modes[i == modeCount - 1 ? 0 : i + 1];
+                        }
+
+                        break;
+                    }
+                }
```

Comment:
I wonder if it wouldn't be easier to understand if the selection of the batch mode would be done more explicitly and directly. Using the array as a middleman might not even be necessary.

Basically a `switch` with explicit state transitions. and avoid the complicated index operations.